### ISSUE TITLE
Add functionality for default layout and settings

### ISSUE THREAD

**Issue by jlaw9** [Murali-group/GraphSpace#289]
In the metadata of an uploaded JSON file, we currently store the graph 'description', 'title', 'tags'. I propose we also allow storing a 'default_layout' option which can be set to any of the available layouts (such as 'cola' or 'spring_embedder'). We would also then need a 'layout_options' dictionary which would store the options the user wants set for the specified layout.

This also leads to another feature in the UI to allow the user to specify a default automatic layout (rather than the current 'default_layout' which stores x and y coord.) and to allow the user to set options for an automatic layout in the UI for a given graph.

**Comment by lohani2280** [comment]
@adbharadwaj @jlaw9  I am a newcomer here. I want to work on this issue. I have installed GraphSpace locally. Can you please elaborate the issue a bit more?

**Comment by adbharadwaj** [comment]
1. You should refer to the [Default Layout](http://manual.graphspace.org/en/latest/Interacting_with_Graphs.html#default-layout) documentation.  

2. Currently a user can save layouts using `Layout Editor` only. Once a user has saved a layout they can set it as default. 

3. But what if a user wants to save an [automatic layout option](http://manual.graphspace.org/en/latest/Interacting_with_Graphs.html#change-layout) as default? Currently they will have to first change the layout. Go to layout editor. Save it and then set it as default. Too many clicks!! :(

The goal is to reduce the number of clicks required to set an automatic layout option as default. There could be two solutions:

- Show the button to set layout as default when an automatic algorithm has been applied. When user sets an automatic algorithm as default, save the layout currently generated layout as a manual layout and then set it as default. (All of this done behind the scene) You need to add this logic to set layout as default button. Problem with this issue is that we are not setting algorithm as default, rather we are saving and setting current layout as default. (Easier)

- Another option is to add dedicated rows for all of the automatic algorithm and then set the layout to  layout id specific to the given algorithm. Problem with this issue is that this will not save the exact layout user us seeing right now. Since, output of a layout is dependent on screen size. Lets say the user uses a different machine with different size next time and they see a different layout, it might be a bad experience. (Moderate - will require SQL knowledge)

@jlaw9 any thoughts? which one would you prefer?

**Comment by jlaw9** [comment]
So my goal was to allow someone who's posting a graph through the API to be able to have an automatic layout applied automatically without needing to click anything.  I think the second point is what would need to be implemented for that to work. I don't think the issue you bring up about not saving the exact layout is a problem because I just want the automatic layout to be applied. If I wanted to keep the positions, then I would save it as a layout. 

I do think the first solution is good to have as well though.

**Comment by tmmurali** [comment]
To implement "allow someone who's posting a graph through the API to be able to have an automatic layout applied automatically without needing to click anything" shouldn't we first add a property to the graph JSON that specifies the name of the layout algorithm, as Jeff suggests in the definition of the issue? 

In the UI, adding a row for each algorithm seems clunky (the second approach suggested by Ted). It will clutter up the UI.

**Comment by jlaw9** [comment]
I agree with @tmmurali. The real reason I opened this issue is for the following scenario:

Suppose I post 100+ graphs and I make them public along with a published manuscript. If someone new to GraphSpace opens one of those graphs, I want to have the cola layout with specific settings applied automatically without requiring me to manually set the default layout for each of those 100+ graphs. 

This should be doable through the API and would require us to add additional properties in either the graph or style JSON to set the automatic layout and options.

Here is an example for [cola](https://github.com/cytoscape/cytoscape.js-cola):
```
{
  'automatic_layout': 'cola',
  'automatic_layout_options': {
    'animate': true,
    'avoidOverlap': true,
    'flow': { 'axis': 'y', 'minSeparation': 30},
    'edgeLength': 150,
    'infinite': true
}
```
